keps/sig-node/4004-deprecate-kube-proxy-version/README.md

```diff
@@ -0,0 +1,265 @@
+
+### Risks and Mitigations
+
+## Design Details
+
+### Versioned API Change: NodeStatus v1 core
+
+Mark the kubeProxyVersion field as Deprecated.
+
+```
+// Deprecated: KubeProxy Version reported by the node.
+KubeProxyVersion string `json:"kubeProxyVersion" protobuf:"bytes,8,opt,name=kubeProxyVersion"`
+```
+
+### NodeStatus Internal Representation
+
+Mark the kubeProxyVersion field as Deprecated.
+
+```
+// Deprecated: KubeProxy Version reported by the node.
+KubeProxyVersion string
+```
+
+### Test Plan
+
+[x ] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+
+##### Unit tests
+
+- Add a test to `TestVersionInfo` in `pkg/kubelet/nodestatus` to see if FeatureGate behaves as expected when it is turned on or off.
+- If DisableNodeKubeProxyVersion FeatureGate is enabled:
+  - `status.nodeInfo.kubeProxyVersion` will be empty.
+- Else:
+  - `status.nodeInfo.kubeProxyVersion` will be not empty .
+
+##### Integration tests
+
+
+- N/A
+
+##### e2e tests
+
+ - N/A
+
+### Graduation Criteria
+
+#### Alpha
+
+- [ ] Created the feature gate, disabled by default.
+- [ ] Started looking for components that might be using the deprecated field.
+
+#### Beta
+
+- [ ] Make sure that we fixed the GCP cloud provider and any other components we found that use the deprecated field.
+- [ ] Make sure that upgrades to the Beta version will work across supported levels of [version skew](https://kubernetes.io/releases/version-skew-policy/).
+
+#### GA
+
+- [ ] No issues reported during two releases.
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: DisableNodeKubeProxyVersion
+  - Components depending on the feature gate: kubelet
+
+###### Does enabling the feature change any default behavior?
+
+Yes, it makes kubelet stop setting `node.status.nodeInfo.kubeProxyVersion`.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Using the featuregate is the only way to enable/disable this feature.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The feature should continue to work just fine.
+
+###### Are there any tests for feature enablement/disablement?
+
+No, we don't need to test what happens when the value of the field changes, because our goal is to make sure no one is looking at the value at all, so no components should notice if the value changes.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
```

Comment:
If there is a component that we didn't know about using `kubeProxyVersion`, then it may fail in some unknown way when the administrator upgrades to a version with that field disabled. It should be possible to just roll back in this case.